.github/workflows/autobuild.yml

```
@@ -122,22 +122,21 @@ jobs:
             base_command:           TARGET_ARCH=armhf ./.github/autobuild/linux_deb.sh
             run_codeql:             false
 
-          - config_name:            MacOS (artifacts)
+          - config_name:            MacOS (artifacts+CodeQL)
             target_os:              macos
             # Stay on 10.15 as long as we use dmgbuild which does not work with 11's hdiutil (?):
             building_on_os:         macos-10.15
             base_command:           QT_VERSION=5.15.2 SIGN_IF_POSSIBLE=1 ./.github/autobuild/mac.sh
-            # Disable CodeQL on mac as it interferes with signing the binaries (signing hangs, see #2563 and #2564)
-            run_codeql:             false
+            # run_codeql will be ignored in the steps below if building a signed image
+            # for Mac, as it causes the signing process to hang. See #2563 and #2564.
+            run_codeql:             true
```

Hm, there might be another approach which could work with less additional logic (although I haven't tried it):
Can't we just disable `run_codeql` based on the existence of the secrets?

Seems like [direct secrets access isn't supported for `if` expressions](https://stackoverflow.com/questions/70249519/how-to-check-if-a-secret-variable-is-empty-in-if-conditional-github-actions), but environment variables should work.

```suggestion
            run_codeql:             ${{ env.MACOS_CERT_ID && 'false' || 'true' }}
```
(i.e. this additionally needs a job-level `env:  MACOS_CERT_ID: ${{ secrets.MACOS_CERT_ID}}`)

There is a downside of this approach though: In repos which "enable" signing (= have the required secrets), but don't have a valid signing config (bad password, missing some other secrets), CodeQL will not run although it could. The original approach from this PR handles this more correctly. I'd argue that this is not relevant though. First, there's likely only one repo (emlyn's) which does proper signing. Second, we mainly care for CodeQL on our jamulussoftware repo.

---

Approaches like this were already tried by @emlynmac without success when developing his recent PR, which got superseded. I'm not sure environment variables are necessarily available at the time the steps are set up, only when they are executed. So I think your suggestion won't work. That is why I looked at having the `setup` phase set an output we could test in an `if` condition on a step, as that is a run-time check.

---

Ah, I had missed https://github.com/emlynmac/jamulus/commit/1bd0a022a1a242ad207d35c9ccd9a60d70633032 (related [CI failure](https://github.com/emlynmac/jamulus/actions/runs/2055012005)). Seems to be related to the matrix not supporting that. Too bad. :(